File: multiStructure/frameworks/js-playwright/pages/checkout/checkoutOverviewPage.js
class CheckoutOverview {
  constructor(page) {
    this.page = page;

    this.title = page.locator('.title');
    this.productPrice = page.locator('.inventory_item_price');
    this.finishButton = page.locator('#finish');
  }

  async getProductPrice() {
    const priceText = await this.productPrice.textContent();
    return parseFloat(priceText.replace('$', ''));
  }

  async getTitleText() {
    return await this.title.first().textContent();
  }

  async clickOnFinishButton() {
    await this.finishButton.click();
  }
}

module.exports = { CheckoutOverview };

File: multiStructure/frameworks/csharp-playwright/Pages/CheckoutOverviewPage.cs
using Microsoft.Playwright;

namespace PlaywrightFramework.Pages;

public class CheckoutOverviewPage
{
    private readonly IPage _page;

    public CheckoutOverviewPage(IPage page)
    {
        _page = page;
    }

    private ILocator Title => _page.Locator(".title").First;
    private ILocator ProductPrice => _page.Locator(".inventory_item_price");
    private ILocator FinishButton => _page.Locator("#finish");

    public async Task<double> GetProductPriceAsync()
    {
        var priceText = await ProductPrice.TextContentAsync();
        return double.Parse(priceText!.Replace("$", ""));
    }

    public async Task<string> GetTitleTextAsync()
    {
        return await Title.TextContentAsync() ?? string.Empty;
    }

    public async Task ClickOnFinishButtonAsync()
    {
        await FinishButton.ClickAsync();
    }
}